docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,412 @@
+transfer some specific information. For example, some attestations follow the
+Diffie–Hellman key exchange protocol to first build a secure channel and
+transfer secret messages (Such as AMD SEV(-ES) pre-attestation).
 
-- `token`
+## `Challenge`
 
-If other requests have been made before this request, in the previous response KBS would return a 
-token to KBC. There are two cases for the contents of this field: 
+If the KBC does not own any KBS generated HTTP Cookie, or if the Cookie validity
+has expired, the KBC can not directly request resources and the KBS respond with
+an error to the KBC requests. The KBC must first go through the attestation
+process and send an initial [`Request`](#request) payload to the KBS. The KBS
+will then respond to it with an attestation challenge described in the following
+format:
+
+```json
+{
+    /* Evidence freshness. */
+    "nonce": "$nonce",
+    /* Extra parameters to support some special HW-TEE attestation. */
+    "extra-params": {}
+}
+```
+> **Note:** the `/*...*/` comments are not valid in JSON, and must not be used.
 
-1. A token is provided, which means a previous attestation was done. If the token is valid, 
-the Challenge/Attestation part can be skipped. If it is not valid, KBS will send a Challenge.
+- `nonce`
 
-2. Left blank, which means the KBC is implictly expecting to go through attestation 
-and KBS will reply with a Challenge.
+The freshness number passed to KBC. KBC needs to place it in the evidence sent
+to the KBS in the next step to prevent replay attacks.
 
 - `extra-params`
 
-In the run-time attestation scenario (TDX, SEV-SNP, SGX), the `extra-params` are not used, so it is kept empty.
-However, for the attestation of some special HW-TEE platforms, this field may be used to transfer some specific information,
-for example, some attestations depend on Diffie–Hellman to build a secure channel and transfer secret messages 
-(Such as the SEV(-ES) pre-attestation).
+The reserved extra parameter field which is used to pass the additional
+information provided by the KBS when some specific HW-TEE needs to be attested.
 
-## Challenge
+## `Attestation`
 
-After KBS receives the request, if the token is found to be empty or expired, KBS will return an 
-attestation challenge to KBC. The payload format is as follows:
+After receiving the attestation challenge, the KBC builds an attestation
+evidence from the HW-TEE platform and organizes it into the following payload:
 
 ```json
 {
-    "challenge": {
-        /* To ensure the freshness of evidence. */
-        "nonce": "",
-        /* Extra parameters to support some special HW-TEE attestation. */
-        "extra-params": {}
-    }
+    /*
+     * A JWK-formatted public key, generated by the KBC running in the HW-TEE.
+     * It is valid until the next time an attestation is required. Its hash must
+     * be included in the HW-TEE evidence and signed by the HW-TEE hardware.
+     */
+    "tee-pubkey": $pubkey
+
```

Comment:
As we discussed earlier, perhaps we need to add a `measurements` field (or another synonymous name) to `Attestation` json payload? Just like this:

```json
{
    "tee-pubkey": $pubkey,
    "measurements": {
        $name: $value,
        ...
    },
    "tee-evidence": {}
}
```

Reply:
cc @slp @sameo